### PR TITLE
Rename composer.json.dist to composer.json

### PR DESCRIPTION
**Pull request by mvorisek**
needed for #9241, for using this project/libs as a composer/Packagist dependency in general

### CODE REVIEW

File: bin/installto.sh [hunk 1]
```diff
@@ -78,8 +78,9 @@ if (strtolower($input) == 'y') {
         }
     }
 
-    foreach (['index.php', 'config/defaults.inc.php', 'composer.json-dist', 'jsdeps.json', 'CHANGELOG.md', 'README.md', 'UPGRADING', 'LICENSE', 'INSTALL'] as $file) {
-        $command = "rsync -a --out-format=%n " . INSTALL_PATH . "$file $target_dir/$file";
+    foreach (['index.php', 'config/defaults.inc.php', 'composer.json.dist', 'jsdeps.json', 'CHANGELOG.md', 'README.md', 'UPGRADING', 'LICENSE', 'INSTALL'] as $file) {
```
Comment (mvorisek) [inline bin/installto.sh:81]: This ensures the `composer.json` is downloaded still to `composer.json.dist` as expected by https://github.com/roundcube/roundcubemail/blob/1.6.5/bin/update.sh#L177 update script.

I couldn't find more places to fix, but please review this PR carefully, there might be more install usecases/docs I am not aware.

File: composer.json [hunk 2]
```diff
@@ -3,7 +3,7 @@
     "description": "The Roundcube Webmail suite",
     "license": "GPL-3.0-or-later",
     "require": {
-        "php": ">=7.3.0",
+        "php": ">=7.3 <8.4",
```
Comment (alecpl) [inline composer.json:6]: I'm not sure about this change.

**Comment by alecpl** [comment]
This will make working directly on source code more complicated. I normally have my composer.json file with additional packages/plugins added. I won't be able to do that. I need to investigate what options there are to fix that. One I found is use of `COMPOSER` env pointing to a different file.

**Comment by mvorisek** [comment]
`COMPOSER` env. var [1] is for root package [2] only. Dependencies require to use `composer.json` strictly. I belive there is no workaround and this PR is needed.

[1] https://getcomposer.org/doc/03-cli.md#composer
[2] https://getcomposer.org/doc/04-schema.md#root-package

File: composer.json (hunk 2 above)
Comment (mvorisek) [reply composer.json]: landed in #9298

File: bin/installto.sh [hunk 3]
```diff
@@ -79,7 +79,8 @@ if (strtolower($input) == 'y') {
     }
 
     foreach (['index.php', 'config/defaults.inc.php', 'composer.json-dist', 'jsdeps.json', 'CHANGELOG.md', 'README.md', 'UPGRADING', 'LICENSE', 'INSTALL'] as $file) {
-        $command = 'rsync -a --out-format=%n ' . INSTALL_PATH . "$file $target_dir/$file";
+        $source_file = $file === 'composer.json-dist' ? 'composer.json' : $file;
```
Comment (mvorisek) [inline bin/installto.sh:82]: This ensures the `composer.json` is downloaded still to `composer.json-dist` as expected by https://github.com/roundcube/roundcubemail/blob/1.6.5/bin/update.sh#L177 update script.

I couldn't find more places to fix, but please review this PR carefully, there might be more install/update usecases/docs I might be not fully aware.

**Comment by alecpl** [comment]
I created https://github.com/composer/composer/issues/11784 but I don't expect it to be implemented soon or at all.

The package I mentioned there is not of a great quality. However, I tested the `COMPOSER` variable and it looks like it might actually work. But it's not an easy decision.

**Comment by mvorisek** [comment]
@alecpl can I ask you to review this PR? Based on https://github.com/roundcube/roundcubemail/pull/9279#issuecomment-1868383175 and https://github.com/composer/composer/issues/11784 the `composer.json` name is the only one name possible to make this project working with composer natively.

**Comment by mvorisek** [comment]
I have rebased this PR. @alecpl, can you please check? According to the sources above, there are no other options, the only supported filename by composer is `composer.json` and the place that needs to be adjusted is the distrubution/upgrade.

**Comment by mvorisek** [comment]
@alecpl, would you mind to look into this PR?

**Pull request merged**